### DISCUSSION TITLE
Standard way to pass custom `-P` argument to starship?

### DISCUSSION THREAD

**Discussion by LunarLambda** [starship/starship#5706] [Q&A] [answered]
Zsh has the concept of named directories (zshexpn(1)), such as `~user` (user's `$HOME`), `~env_var` (if `$env_var` starts with a `/`), and custom ones set via `hash -d`.

In Zsh prompt expansion (zshmisc(1)), the `%~` specifier can be used to show the current directory, shortened to this tilde-syntax. I would like to use this in starship.

Through experimentation I figured out I can dump `starship init zsh` out to a file, then change the `PROMPT` to include `-P "${(%):-%~}"`, then source that, and it seems to work without issue. Is there a standard way to do this? Or is this the intended way to include shell-specific customizations?

**Comment by davidkna** [comment]
`$PWD` would also overwrite it, but that may also have other side effects. There is currently no official method for adding custom arguments to the starship binary.

**Reply by LunarLambda** [reply]
Yeah messing with PWD isn't a good idea I think.

Is there any problem with my workaround (other than potentially needing to re-do it when updating starship)? As far as I can tell the output of `starship init` is always the same.

**Reply by davidkna** [reply]
The init code sometimes changes between versions, but not very often.

**Reply by LunarLambda** [reply]
yeah, I meant between invocations in the same version. The output isn't dynamic

**Reply by davidkna** [reply, marked as answer]
Apart from the starship binary path the output isn't dynamic, and that's unlikely to change in the near future.

**Reply by LunarLambda** [reply]
I see. I'll stick to that for the time being then. Thank you
